LogMonitor/src/LogMonitor/JsonFileParser.cpp

```diff
@@ -145,6 +145,107 @@ JsonFileParser::ParseSpecialCharacter(int Ch)
     return (static_cast<wchar_t>(Ch));
 }
 
+///
+/// Parses a real number (positive or negative integers or decimals) passed as a number (eg. 10) or string (eg. "10").
+/// It also parses infinity values (eg.: Inf, -Inf, INFINITY, INFINITY)
+///
+/// \return A double value.
+///
+const std::double_t &
+JsonFileParser::ParseNumericValue()
+{
+    size_t offset = 0;
+    int ch = PeekNextCharacter(offset);
+
+    if (ch != '"')
+    {
+        return ParseNumber();
+    }
+    else
+    {
+        // Convert string numerical value or infinity to double
+        std::wstring parsedStringValue = ParseStringValue();
+        try
+        {
+            // Convert from wide string to narrow string
+            // const std::string s(parsedValue.begin(), parsedValue.end());
+
+            //// convert string to double
+            m_doubleValue = std::stod(parsedStringValue.c_str());
+            return m_doubleValue;
+        }
+        catch (...)
+        {
+            throw std::invalid_argument("JsonFileParser: Invalid value. Expected number or infinity");
+        }
+    }
+}
+
+///
+/// Parses a real number (positive or negative integers or decimals) passed as a number
+///
+/// \return A double value.
+///
+const std::double_t&
+JsonFileParser::ParseNumber()

```

Comment:
In addition to the value "INFINITY" and "Inf", can we also decode the Infinity symbol as well?

Example of configuration in JSON file:
![image](https://github.com/microsoft/windows-container-tools/assets/28774968/9d70d00b-d7e0-4805-802e-fd988ada0880)


Current behaviour (the tool interprets the infinity symbol as a zero):
![image](https://github.com/microsoft/windows-container-tools/assets/28774968/d7d0f856-67d9-43fa-8d24-94a30ba515d4)